osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java

```diff
@@ -304,7 +308,24 @@ public PageOfUsers queryUsers(long offset, long limit, String filter) {
 
     @Override
     public OsisS3Credential createS3Credential(String tenantId, String userId) {
-        throw new NotImplementedException();
+        try {
+            logger.info("Create S3 Credential request received:: tenant ID:{}, user ID:{}",
+                    tenantId, userId);
+
+            Credentials tempCredentials = getCredentials(tenantId);
+            final AmazonIdentityManagement iamClient = vaultAdmin.getIAMClient(tempCredentials, appEnv.getRegionInfo().get(0));
+
+            OsisS3Credential credential =  createOsisCredential(tenantId, userId, null, null, iamClient);
+
+            logger.trace("Create S3 Credential response with secret:{}", new Gson().toJson(credential));
```

Comment:
I researched about this, we should not log secret key even in trace logs.

More info in the summary of this ticket: https://scality.atlassian.net/browse/S3C-4366

Reply:
As discussed for older logs(already committed), we can clean up later.

Reply:
fixed in f250542fa2138cb24ab0e2fd1ceeb97abef1750d